docs/tutorials/data_provider.md

```diff
@@ -62,6 +62,83 @@ impl AdditiveIdentity {
 }
 ```
 
+## Loading Additional Data at Runtime
+
+A key feature of ICU4X is the ability to download data dynamically. A key use case is to add additional locales at runtime.
+
+Dynamic data loading can currently be performed in userland. A future core library API may provide this functionality; please submit feedback in [#2985](https://github.com/unicode-org/icu4x/issues/2985).
```

Comment:
nit: "userland", to me, usually means non-kernel-privileged code: should we say "user code" instead?